File: deltafi-ui/src/mocks/api/v2/roles.ts
/*
   DeltaFi - Data transformation and enrichment platform

   Copyright 2021-2025 DeltaFi Contributors <deltafi@deltafi.org>

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/

const generateData = () => {
  return [
    {
      id: 1,
      name: "Admin",
      permissions: [
        "Admin"
      ],
      createdAt: "2022-11-30T19:55:51.760Z",
      updatedAt: "2022-11-30T19:55:51.760Z"
    },
    {
      id: 2,
      name: "Ingress Only",
      permissions: [
        "DeltaFileIngress"
      ],
      createdAt: "2022-11-30T19:55:51.763Z",
      updatedAt: "2022-11-30T19:55:51.763Z"
    },
    {
      id: 3,
      name: "Read Only",
      permissions: [
        "DashboardView",
        "DeletePolicyRead",
        "DeltaFileContentView",
        "DeltaFileMetadataView",
        "FlowView",
        "MetricsView",
        "PluginsView",
        "SnapshotRead",
        "StatusView",
        "SystemPropertiesRead",
        "UIAccess",
        "VersionsView"
      ],
      createdAt: "2022-11-30T19:55:51.765Z",
      updatedAt: "2022-11-30T19:55:51.765Z"
    }
  ]
};

export default generateData();

File: deltafi-auth/routes/roles.rb
#
#    DeltaFi - Data transformation and enrichment platform
#
#    Copyright 2021-2025 DeltaFi Contributors <deltafi@deltafi.org>
#
#    Licensed under the Apache License, Version 2.0 (the "License");
#    you may not use this file except in compliance with the License.
#    You may obtain a copy of the License at
#
#        http://www.apache.org/licenses/LICENSE-2.0
#
#    Unless required by applicable law or agreed to in writing, software
#    distributed under the License is distributed on an "AS IS" BASIS,
#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#    See the License for the specific language governing permissions and
#    limitations under the License.
#

# frozen_string_literal: true

class AuthApi < Sinatra::Application
  get '/roles' do
    authorize! :RoleRead

    Role.all.map(&:to_api).to_json
  end

  get '/roles/:id' do
    authorize! :RoleRead

    id = params['id'].to_i
    role = Role[id]

    if role.nil?
      status 404
      return error("Role with ID #{id} not found.")
    end

    role.to_api.to_json
  end

  post '/roles' do
    authorize! :RoleCreate

    role = Role.new(read_body)

    unless role.valid?
      status 400
      return { validation_errors: role.errors }.to_json
    end

    role.save

    role.to_api.to_json
  end

  put '/roles/:id' do
    authorize! :RoleUpdate

    id = params['id'].to_i
    role = Role[id]

    if role.nil?
      status 404
      return error("Role with ID #{id} not found.")
    end

    role.update(read_body)

    unless role.valid?
      status 400
      return { validation_errors: role.errors }.to_json
    end

    role.save
    audit("updated role #{role.name}")

    Role[id].to_api.to_json
  end

  delete '/roles/:id' do
    authorize! :RoleDelete

    id = params['id'].to_i
    if id == 1
      status 403
      return error('Unable to delete the admin role.')
    end

    role = Role[id]

    if role.nil?
      status 404
      return error("Role with ID #{id} not found.")
    end

    role.destroy
    audit("deleted role #{role.name}")

    return role.to_api.to_json
  end
end
